Gems/AtomTressFX/Assets/Shaders/HairLightTypes.azsli

```diff
@@ -0,0 +1,633 @@
+enum class HairLightingModel {GGX, Marschner, Kajiya};
+option HairLightingModel o_hairLightingModel = HairLightingModel::Marschner;
+
+//==============================================================================
+float3 GetSpecularLighting(Surface surface, LightingData lightingData, const float3 lightIntensity, const float3 dirToLight)
+{
+    float3 specular = float3(1, 0, 1);  // purple - error color
+    if (o_hairLightingModel == HairLightingModel::GGX)
+    {
+        specular = SpecularGGX(lightingData.dirToCamera, dirToLight, surface.normal, surface.specularF0, lightingData.NdotV, surface.roughnessA2, lightingData.multiScatterCompensation);
+    }
+    else if(o_hairLightingModel == HairLightingModel::Marschner)
+    {
+        specular = HairMarschnerBSDF(surface, lightingData, dirToLight);
+    }
+
+    return specular * lightIntensity;
+}
+
+float3 GetHairBackLighting(Surface surface, LightingData lightingData, float3 lightIntensity, float3 dirToLight, float shadowRatio)
+{
+    if (o_hairLightingModel == HairLightingModel::GGX)
+    {
+        float3 result = float3(0.0, 0.0, 0.0);
+        float thickness = 0.0;
+        float4 transmissionParams = surface.transmission.transmissionParams;
+
+        // Thin object mode, using thin-film assumption proposed by Jimenez J. et al, 2010, "Real-Time Realistic Skin Translucency"
+        // http://www.iryoku.com/translucency/downloads/Real-Time-Realistic-Skin-Translucency.pdf
+
+        result = shadowRatio ?
+            float3(0.0, 0.0, 0.0) :
+            TransmissionKernel(surface.transmission.thickness * transmissionParams.w, rcp(transmissionParams.xyz)) *
+            saturate(dot(-surface.normal, dirToLight)) * lightIntensity * shadowRatio;
+
+        return result;
+    }
+    else // if ((o_hairLightingModel == HairLightingModel::Marschner) || (o_hairLightingModel == HairLightingModel::Kajiya))
+    {
+        return float3(0.0f, 0.0f, 0.0f);
+    }
+
+    return float3(1.0f, 0.0f, 0.0f);
+}
+
+//! Simple Lambertian BRDF
+float3 HairDiffuseLambertian(float3 albedo, float3 normal, float3 dirToLight)
+{
+    float NdotL = saturate(dot(normal, dirToLight));
+    return albedo * NdotL * INV_PI;
+} 
+
+// Replacing the generic Diffuse and Specular methods in StandardLighting.azsli 
+// and removing the regular usage of clear coat second lobe energy distribution.
+float3 GetDiffuseLighting(Surface surface, LightingData lightingData, float3 lightIntensity, float3 dirToLight)
+{
+    float3 diffuse = float3(0, 1, 0);   // Green - error color
+
+    if (o_hairLightingModel == HairLightingModel::GGX)
+    {
+        // Notice that addition of the response (1-F) here
+        diffuse = HairDiffuseLambertian(surface.albedo, surface.normal, dirToLight) * lightingData.diffuseResponse;
+    }
+    else if (o_hairLightingModel == HairLightingModel::Marschner)
+    {
+        return float3(0.0f, 0.0f, 0.0f);
+    }
```

Comment:
Same as my previous comment, can you add a small comment here saying why there's no HairLightingModel::Kajiya option and where that is handled?

Reply:
done